src/wp-includes/webfonts-api/providers/class-wp-webfonts-google-provider.php

```diff
@@ -0,0 +1,147 @@
+<?php
+/**
+ * Webfonts API: Google Fonts provider.
+ *
+ * @package    WordPress
+ * @subpackage WebFonts
+ * @since      5.9.0
+ */
+
+/**
+ * Webfonts API provider for Google Fonts.
+ */
+class WP_Webfonts_Google_Provider extends WP_Webfonts_Provider {
+
+	/**
+	 * The provider's unique ID.
+	 *
+	 * @since 5.9.0
+	 *
+	 * @var string
+	 */
+	protected $id = 'google';
+
+	/**
+	 * An array of URLs to preconnect to.
+	 *
+	 * @since 5.9.0
+	 * @var array
+	 */
+	protected $preconnect_urls = array(
+		array(
+			'href'        => 'https://fonts.gstatic.com',
+			'crossorigin' => true,
+		),
+		array(
+			'href'        => 'https://fonts.googleapis.com',
+			'crossorigin' => false,
+		),
+	);
+
+	/**
+	 * The provider's root URL.
+	 *
+	 * @since 5.9.0
+	 * @var string
+	 */
+	protected $root_url = 'https://fonts.googleapis.com/css2';
+
+	/**
+	 * An array of API parameters which will not be added to the @font-face.
+	 *
+	 * @since 5.9.0
+	 * @var array
+	 */
+	protected $invalid_parameters = array(
+		'subset',
+		'text',
+		'effect',
+	);
```

Comment:
>The "invalid parameters" was a bad name for that var... The Google-fonts API supports some extra params that don't generate any css themselves, so they are not exactly part of font-face... But they change the way the API behaves. They are edge-cases though and I think we can commit them from v1.

Okay so that are optional and should not be stripped out of the webfont schema configuration. Right now they are being stripped out as they are part of the schema properties.

Sounds like the API's Validator will need a way to grab extra params from the provider, if it exists.